test/uk/gov/hmrc/uknwauthcheckerapi/services/ValidationServiceSpec.scala

```diff
@@ -88,41 +91,112 @@ class ValidationServiceSpec extends BaseSpec {
+          val json = Json.toJson(noEorisRequest.request)
+
+          val expectedResponse =
+            ValidationDataRetrievalError(
+              JsError(
+                Seq((JsPath \ JsonPaths.eoris, Seq(JsonValidationError("The request payload must contain between 1 and 3000 EORI entries"))))
+              )
+            )
+
+          val request = fakeRequestWithJsonBody(json)
+
+          val response = service.validateRequest(request)
+
+          response shouldBe Left(expectedResponse)
+        }
       }
     }
 
-    "return JsError when AuthorisationRequest has no Eoris" in {
-      forAll { (noEorisRequest: NoEorisAuthorisationRequest) =>
-        val json = Json.toJson(noEorisRequest.request)
+  }
+}
+
+class ValidationServiceOverwrittenMaxEoriSpec extends BaseSpec {
 
-        val expectedResponse =
-          ValidationDataRetrievalError(
-            JsError(
-              Seq((JsPath \ JsonPaths.eoris, Seq(JsonValidationError(ApiErrorMessages.invalidEoriCount))))
+  private val maxEori = 10
+  private val configValue: Map[String, Int] = Map("microservice.services.self.eoriMax" -> maxEori)
+
+  override def fakeApplication(): Application =
+    GuiceApplicationBuilder()
+      .configure(additionalAppConfig ++ configValue)
+      .overrides(moduleOverrides)
+      .build()
+
+  private val service = new ValidationService(minMaxValues)
+
+  "validationRequest with overwritten config" should {
+
+    "must return JSError with different content depending on maxEori" when {
+
+      "AuthorisationRequest has too many Eoris" in {
+        forAll { (tooManyEorisRequest: TooManyEorisAuthorisationRequest) =>
+          val json = Json.toJson(tooManyEorisRequest.request)
+
+          val expectedResponse =
+            ValidationDataRetrievalError(
+              JsError(
+                Seq(
+                  (
+                    JsPath \ JsonPaths.eoris,
+                    Seq(JsonValidationError(s"The request payload must contain between 1 and $maxEori EORI entries"))
+                  )
+                )
+              )
             )
-          )
 
-        val request = fakeRequestWithJsonBody(json)
+          val request = fakeRequestWithJsonBody(json)
 
-        val response = service.validateRequest(request)
+          val response = service.validateRequest(request)
 
-        response shouldBe Left(expectedResponse)
+          response shouldBe Left(expectedResponse)
+        }
+      }
+
+      "when AuthorisationRequest has no Eoris" in {
+        forAll { (noEorisRequest: NoEorisAuthorisationRequest) =>
+          val json = Json.toJson(noEorisRequest.request)
+
+          val expectedResponse =
+            ValidationDataRetrievalError(
+              JsError(
+                Seq(
+                  (
+                    JsPath \ JsonPaths.eoris,
+                    Seq(JsonValidationError(s"The request payload must contain between 1 and $maxEori EORI entries"))
```

Comment:
Can you utilise the ApiErrorMessages. We don't want literals where we can avoid it.